x/ibc/02-client/keeper/keeper.go

```diff
@@ -134,105 +109,30 @@ func (k Keeper) IterateClients(ctx sdk.Context, cb func(types.State) bool) {
 }
 
 // GetAllClients returns all stored light client State objects.
-func (k Keeper) GetAllClients(ctx sdk.Context) (states []types.State) {
-	k.IterateClients(ctx, func(state types.State) bool {
+func (k Keeper) GetAllClients(ctx sdk.Context) (states []exported.ClientState) {
+	k.IterateClients(ctx, func(state exported.ClientState) bool {
 		states = append(states, state)
 		return false
 	})
 	return states
 }
 
-// GetCommitter will get the Committer of a particular client at the oldest height
-// that is less than or equal to the height passed in
-func (k Keeper) GetCommitter(ctx sdk.Context, clientID string, height uint64) (exported.Committer, bool) {
-	store := ctx.KVStore(k.storeKey)
-
-	var committer exported.Committer
-
-	// TODO: Replace this for-loop with a ReverseIterator for efficiency
-	for i := height; i > 0; i-- {
-		bz := store.Get(types.KeyCommitter(clientID, i))
-		if bz != nil {
-			k.cdc.MustUnmarshalBinaryLengthPrefixed(bz, &committer)
-			return committer, true
-		}
-	}
-	return nil, false
-}
-
-// SetCommitter sets a committer from a particular height to
-// a particular client
-func (k Keeper) SetCommitter(ctx sdk.Context, clientID string, height uint64, committer exported.Committer) {
-	store := ctx.KVStore(k.storeKey)
-	bz := k.cdc.MustMarshalBinaryLengthPrefixed(committer)
-	store.Set(types.KeyCommitter(clientID, height), bz)
-}
-
 // State returns a new client state with a given id as defined in
 // https://github.com/cosmos/ics/tree/master/spec/ics-002-client-semantics#example-implementation
-func (k Keeper) initialize(ctx sdk.Context, clientID string, consensusState exported.ConsensusState) types.State {
-	clientState := types.NewClientState(clientID)
-	k.SetConsensusState(ctx, clientID, consensusState)
-	return clientState
-}
-
-// freeze updates the state of the client in the event of a misbehaviour
-func (k Keeper) freeze(ctx sdk.Context, clientState types.State) (types.State, error) {
-	if clientState.Frozen {
-		return types.State{}, sdkerrors.Wrap(errors.ErrClientFrozen, clientState.ID)
+func (k Keeper) initialize(
+	ctx sdk.Context, clientID string, clientType exported.ClientType,
+	consensusState exported.ConsensusState,
+) (exported.ClientState, error) {
+	var clientState exported.ClientState
+	height := uint64(ctx.BlockHeight())
+
+	switch clientType {
+	case exported.Tendermint:
+		clientState = tendermint.NewClientState(clientID, height)
+	default:
+		return nil, types.ErrInvalidClientType
 	}
 
-	clientState.Frozen = true
+	k.SetConsensusState(ctx, clientID, height, consensusState)
```

Comment:
I don't think this will uniformly work, e.g. with solo machine clients, which do not have a per-height consensus state.

I suggest using `initialise` instead as defined in ICS 2/6/7 (https://github.com/cosmos/ics/pull/348).